### PR TITLE
External resource model

### PR DESCRIPTION
**Pull request by AlineRibeiro**
Hi Steph and Gui,

This is my pull request for ticket 1.Build a model for resources and call it external_resource.

Thank you!

### CODE REVIEW

File: app/models/external_resource.rb [hunk 1]
```diff
@@ -0,0 +1,7 @@
+# frozen_string_literal: true
+
+class ExternalResource < ApplicationRecord
+  validates :name, presence: true
+  validates :link, presence: true, uniqueness: true
+  validates :languages, presence: true
```
Comment (guialbuk) [inline app/models/external_resource.rb]: You can group attributes per validation type:

```suggestion
  validates :languages, :link, :name, presence: true
  validates :link, uniqueness: true
```

File: db/schema.rb [hunk 2]
```diff
@@ -68,6 +68,17 @@
...
 
+  create_table "external_resources", force: :cascade do |t|
+    t.string "name"
+    t.string "link"
+    t.string "tags", default: [], null: false, array: true
+    t.string "languages", default: [], null: false, array: true
+    t.boolean "published", default: false
+    t.datetime "created_at", precision: 6, null: false
+    t.datetime "updated_at", precision: 6, null: false
+    t.index ["link"], name: "index_external_resources_on_link", unique: true
```
Comment (guialbuk) [inline db/schema.rb:79]: 🔥 🧯

File: spec/factories/external_resources.rb [hunk 3]
```diff
@@ -0,0 +1,9 @@
+FactoryBot.define do
+  factory :external_resource do
+    name { 'Test resource name' }
+    link { 'www.resourcelink.com' }
```
Comment (guialbuk) [inline spec/factories/external_resources.rb:4]: As `link` must be unique, it's better to generate unique values for it.

**Review by guialbuk — APPROVED** [review]

File: app/models/external_resource.rb (hunk 1 above)
Comment (AlineRibeiro) [reply app/models/external_resource.rb]: Thank you

**Review by S-Warmenhoven — APPROVED** [review]
Looks good to me! Just a question (since I haven't done testing in a while), does the test "is_expected.to validate..." automatically test both scenarios where, for example, the link is unique and also not unique, even though our factory only has one example?  I know this is more of a refresher for myself than a review of your code, sorry.

**Comment by AlineRibeiro** [comment]
> Looks good to me! Just a question (since I haven't done testing in a while), does the test "is_expected.to validate..." automatically test both scenarios where, for example, the link is unique and also not unique, even though our factory only has one example? I know this is more of a refresher for myself than a review of your code, sorry.

My understanding is that, behind the scenes, rspec and shoulda-matchers create an instance of that class using the provided link by the factory, and then once the instance is created, it tries to create another instance with the very same link. The second instance should fail. If it fails, then it means that our uniqueness validation works, and our test passes.

**Pull request merged**